### PR TITLE
fix: fail gracefully when no cache headers are available

### PR DESCRIPTION
**Pull request by lahirumaramba**
Resolves: #621

Avoid throwing errors when parsing cache control headers for public keys and set the cache duration to 0 (equivalent to no cache) when the cache headers cannot be parsed.

### CODE REVIEW

File: auth/token_verifier.go [hunk 1]
```diff
@@ -476,19 +473,20 @@ func parsePublicKey(kid string, key []byte) (*publicKey, error) {
 	return &publicKey{kid, pk}, nil
 }
 
-func findMaxAge(resp *http.Response) (*time.Duration, error) {
+func findMaxAge(resp *http.Response) *time.Duration {
+	duration := time.Duration(0) * time.Second
```
Comment (DellaBitta) [inline auth/token_verifier.go:477]: Can this line go with the default return value at the bottom of the function?

**Review by hsubox76 — APPROVED** [review]

File: auth/token_verifier.go (hunk 1 above)
Comment (lahirumaramba) [reply auth/token_verifier.go]: Thanks! Moved the default value to the bottom

**Review by jonathanedey — APPROVED** [review]
LGTM

**Review by DellaBitta — APPROVED** [review]

**Review by Xiaoshouzi-gh — APPROVED** [review]
LGTM.  Thank you so much for the quick response Lahiru!

qq - r we able to test this against a real response?

**Comment by lahirumaramba** [comment]
> LGTM. Thank you so much for the quick response Lahiru!
> 
> qq - r we able to test this against a real response?

Yeah, the integrations tests are all green!

**Pull request merged**